src/storage/src/cell_based_row_deserializer.rs

```diff
@@ -155,13 +163,33 @@ impl<Desc: Deref<Target = ColumnDescMapping>> CellBasedRowDeserializer<Desc> {
         Ok(result)
     }
 
+    /// When we encounter a new key, we can be sure that the previous row has been fully
+    /// deserialized. Then we return the key and the value of the previous row.
+    pub fn deserialize(
+        &mut self,
+        raw_key: impl AsRef<[u8]>,
+        cell: impl AsRef<[u8]>,
+    ) -> Result<Option<(VirtualNode, Vec<u8>, Row)>> {
+        self.deserialize_inner::<true>(raw_key, cell)
+    }
+
+    // TODO: remove this once we refactored lookup in delta join with cell-based table
```

Comment:
I still don't quite understand this workaround. I see in the code we already `.extend(DEFAULT_VNODE)` when calculate the prefix of scan key in lookup executor. Why still call `deserialize_without_vnode` instead of `deserialize`?

Reply:
This is actually a pk prefix scan with `Keyspace.append::scan` instead of `Keyspace::scan`, so the vnode part and the pk prefix of the key passed into the deserializer are truncated.